Installation of curl fails on 'one or more libs available at link-time are not available run-time'

## Description

Continuation of my comments in issue #1501. Was hoping it was fixed after the merge of the PR, but I run into a different issue now where the installation of curl fails on 'one or more libs available at link-time are not available run-time'.

I already have a curl installed, likely via the 'dev_install' command just after putting my Chromebook in developer mode, which is installed in /usr/bin instead of /user/local/bin where curl installs:
```
chronos@localhost ~/Downloads/tools $ which curl
/usr/bin/curl

chronos@localhost ~/Downloads/tools $ curl --version
curl 7.55.1 (x86_64-cros-linux-gnu) libcurl/7.55.1 OpenSSL/1.0.2l zlib/1.2.11
Release-Date: 2017-08-14
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smtp smtps telnet tftp 
Features: Largefile NTLM SSL libz TLS-SRP UnixSockets HTTPS-proxy
```
I was trying to install it via curl too, since I was trying to install R from source and it couldn't find curl.
If this is the root cause and if I shouldn't be trying to install it again I'd appreciate suggestions how you deal with these duplicate packages.

This is the command and (abbreviated) output I get when installing curl:
```
chronos@localhost ~/Downloads/tools $ crew install curl
curl: Command line tool and library for transferring data with URLs.
https://curl.haxx.se/
version 7.57.0-1
No precompiled binary available for your platform, downloading source...
Archive downloaded
Unpacking archive, this may take awhile...
gcc (GCC) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... yes
checking whether to enable debug build options... no
checking whether to enable compiler optimizer... (assumed) yes
```
[...]
```
checking for time.h... (cached) yes
checking for sys/socket.h... (cached) yes
checking for struct timeval... yes
checking run-time libs availability... failed
configure: error: one or more libs available at link-time are not available run-time. Libs used at link-time: -lrtmp -lssh2  -lssl -lcrypto -lssl -lcrypto -lz 
curl failed to install: `./configure --prefix=/usr/local --libdir=/usr/local/lib64 --disable-debug --disable-dependency-tracking --disable-static` exited with 1
```

## Versions and system information
Chromebrew
```
chronos@localhost ~/Downloads/tools $ crew --version
0.4.3
```
ChromeOS
```
chronos@localhost ~/Downloads/tools $ cat /etc/lsb-release
CHROMEOS_ARC_ANDROID_SDK_VERSION=25
CHROMEOS_ARC_VERSION=4446936
CHROMEOS_AUSERVER=https://tools.google.com/service/update2
CHROMEOS_BOARD_APPID={01906EA2-3EB2-41F1-8F62-F0B7120EFD2E}
CHROMEOS_CANARY_APPID={90F229CE-83E2-4FAF-8479-E368A34938B1}
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_APPID={01906EA2-3EB2-41F1-8F62-F0B7120EFD2E}
CHROMEOS_RELEASE_BOARD=eve-signed-mpkeys
CHROMEOS_RELEASE_BRANCH_NUMBER=77
CHROMEOS_RELEASE_BUILDER_PATH=eve-release/R62-9901.77.0
CHROMEOS_RELEASE_BUILD_NUMBER=9901
CHROMEOS_RELEASE_BUILD_TYPE=Official Build
CHROMEOS_RELEASE_CHROME_MILESTONE=62
CHROMEOS_RELEASE_DESCRIPTION=9901.77.0 (Official Build) stable-channel eve 
CHROMEOS_RELEASE_NAME=Chrome OS
CHROMEOS_RELEASE_PATCH_NUMBER=0
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_VERSION=9901.77.0
DEVICETYPE=CHROMEBOOK
GOOGLE_RELEASE=9901.77.0
```
Model: Pixelbook
Architecture:
```
chronos@localhost ~/Downloads/tools $ uname -m
x86_64
```

## Additional information

The stuff that my current curl depends on (I think):
```
chronos@localhost ~/Downloads/tools $ ldd /usr/bin/curl
        linux-vdso.so.1 (0x00007ffc739f6000)
        libcurl.so.4 => /usr/lib64/libcurl.so.4 (0x00007fbc3e3de000)
        libz.so.1 => /usr/local/lib64/libz.so.1 (0x00007fbc3e00c000)
        libc.so.6 => /usr/local/lib64/libc.so.6 (0x00007fbc3dc61000)
        libssl.so.1.0.0 => /usr/local/lib64/libssl.so.1.0.0 (0x00007fbc3d9ef000)
        libcrypto.so.1.0.0 => /usr/local/lib64/libcrypto.so.1.0.0 (0x00007fbc3d599000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fbc3e227000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fbc3d395000)
``` 

Crew searches for what I think the needed libraries are, so you can see what I have installed:
```
chronos@localhost ~/Downloads/tools $ crew search rtmp
(i) rtmpdump: rtmpdump is a toolkit for RTMP streams.
chronos@localhost ~/Downloads/tools $ crew search ssh2
(i) libssh2: libssh2 is a client-side C library implementing the SSH2 protocol.
chronos@localhost ~/Downloads/tools $ crew search ssl
(i) openssl: OpenSSL is an open source project that provides a robust, commercial-grade, and full-featured toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols.
chronos@localhost ~/Downloads/tools $ crew search crypto
libgcrypt: Libgcrypt is a general purpose cryptographic library originally based on code from GnuPG.
nettle: Nettle is a cryptographic library that is designed to fit easily in more or less any context: In crypto toolkits for object-oriented languages (C++, Python, Pike, ...), in applications like LSH or GNUPG, or even in kernel space.
Package template.rb.template not found. :(
chronos@localhost ~/Downloads/tools $ crew search lz
libslz: SLZ is a fast and memory-less stream compressor which produces an output that can be decompressed with zlib or gzip.
lz4: LZ4 is lossless compression algorithm, providing compression speed at 400 MB/s per core (0.16 Bytes/cycle).
lzip: Lzip is a lossless data compressor with a user interface similar to the one of gzip or bzip2.
lzo: LZO is a portable lossless data compression library written in ANSI C.
lzop: lzop is a file compressor which is very similar to gzip.
scrollz: ScrollZ is an advanced IRC client based on ircII.
```

I think we got there! 😃 
Followed your instructions (using a new crosh/bash session in between), but ran into the following error when installing curl, or actually its dependency rtmpdump:
```
chronos@localhost / $ crew install curl
curl: Command line tool and library for transferring data with URLs.
https://curl.haxx.se/
version 7.57.0-1
The following packages also need to be installed: 
openssl rtmpdump 
Do you agree? [Y/n] 
Proceeding...
openssl: OpenSSL is an open source project that provides a robust, commercial-grade, and full-featured toolkit for the Transport Layer Security (TLS) and Secure Sockets Layer (SSL) protocols.
https://www.openssl.org/
version 1.0.2l-1
Precompiled binary available, downloading...
Archive downloaded
Unpacking archive, this may take awhile...
Installing...
Performing post-install...
rtmpdump: rtmpdump is a toolkit for RTMP streams.
https://rtmpdump.mplayerhq.hu/
version fa8646da
No precompiled binary available for your platform, downloading source...
Archive downloaded
Unpacking archive, this may take awhile...
gcc (GCC) 4.9.4
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Building from source, this may take a while...
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\"   -O2   -c -o rtmpdump.o rtmpdump.c
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\"   -O2   -c -o rtmpgw.o rtmpgw.c
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\"   -O2   -c -o thread.o thread.c
make[1]: Entering directory '/usr/local/tmp/crew/fa8646daeb19dfd12c181f7d19de708d623704c0.tar.gz.dir/rtmpdump-fa8646d/librtmp'
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\" -DUSE_OPENSSL  -O2 -fPIC   -c -o rtmp.o rtmp.c
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\"   -O2   -c -o rtmpsrv.o rtmpsrv.c
rtmp.c: In function ‘RTMP_ReadPacket’:
rtmp.c:3555:7: warning: variable ‘didAlloc’ set but not used [-Wunused-but-set-variable]
   int didAlloc = FALSE;
       ^
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\"   -O2   -c -o rtmpsuck.o rtmpsuck.c
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\" -DUSE_OPENSSL  -O2 -fPIC   -c -o log.o log.c
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\" -DUSE_OPENSSL  -O2 -fPIC   -c -o amf.o amf.c
gcc -Wall   -DRTMPDUMP_VERSION=\"v2.4\" -DUSE_OPENSSL  -O2 -fPIC   -c -o hashswf.o hashswf.c
hashswf.c:69:18: fatal error: zlib.h: No such file or directory
 #include <zlib.h>
                  ^
compilation terminated.
make[1]: *** [<builtin>: hashswf.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/usr/local/tmp/crew/fa8646daeb19dfd12c181f7d19de708d623704c0.tar.gz.dir/rtmpdump-fa8646d/librtmp'
make: *** [Makefile:76: librtmp/librtmp.a] Error 2
rtmpdump failed to install: `make -j4 V=0` exited with 2
```

So I searched crew for zlib, found zlibpkg and installed that one. After that the installation of rtmpdump and curl completed successfully and the curl version also actually works!

Could it be that zlib is not documented as dependency for rtmpdump? Could this explain the earlier failures too?
Thanks for all the help and the amazing software.

--- a/packages/rtmpdump.rb
+++ b/packages/rtmpdump.rb
@@ -3,7 +3,7 @@ require 'package'
 class Rtmpdump < Package
   description 'rtmpdump is a toolkit for RTMP streams.'
   homepage 'https://rtmpdump.mplayerhq.hu/'
-  version 'fa8646da'
+  version 'fa8646dae'
   source_url 'https://git.ffmpeg.org/gitweb/rtmpdump.git/snapshot/fa8646daeb19dfd12c181f7d19de708d623704c0.tar.gz'
   source_sha256 'dba4d4d2e1c7de6884b01d98194b83cab6784669089fa3c919152087a3a38fd2'
 
@@ -11,6 +11,8 @@ class Rtmpdump < Package
   })
   binary_sha256 ({
   })
+
+  depends_on 'zlibpkg'
 
   def self.build
     system "sed -i 's,prefix=/usr/local,prefix=#{CREW_PREFIX},' Makefile"

